chef/cookbooks/crowbar/recipes/prepare-upgrade-scripts.rb

```diff
@@ -120,7 +120,7 @@
   group "root"
   action :create
   variables(
-    needs_block_migrate: nova[:nova][:use_shared_instance_storage] ? false : true
+    needs_block_migrate: nova.nil? || (nova[:nova][:use_shared_instance_storage] ? false : true)
```

Comment:
I think the failure of the `nova` being nil is because the resource is first compiled and then converged, and in that moment (converge) is when the `only_if` is checked, but the evaluation of `nova` has already been done in the compile phase

You can work around this by using lazy attributes: https://docs.chef.io/resource_common.html#lazy-evaluation


And I agree with @skazi0 here, the `(nova[:nova][:use_shared_instance_storage] ? false : true)` is kind of weird and out of place.